raspberry/main.py

```diff
@@ -0,0 +1,416 @@
+            print(f"Something wrong with the ToF, maximum number of consecutive altitude outliers recorded: {num_consecutive_altitude_outliers_count_thus_far[0]}."
+                  "".format("\n   **It is recommended to use manual mode in this situation**." if use_altitude_hold[0] else ""))
+
+        is_drone_flying_close_to_ground[0] = last_valid_altitude_measurement_mts[0] < min_altitude_to_activate_AltiHold_mts[0]
+
+        if is_drone_flying_close_to_ground[0]:
+            z_hat[0] = current_altitude_snap_shot_mts[0] if temp == - \
+                1 else temp/1000
+            print_debug(
+                f"Cannot activate altitude hold. Drone is flying close to the ground at {last_valid_altitude_measurement_mts[0]/1000} mts < {min_altitude_to_activate_AltiHold_mts[0]} mts.")
+            if is_KF_ran_atleast_once[0]:
+                kf.reset()
+        else:
+            x_est = kf.update(Tref_t[0], euler[1], euler[2],
+                              np.nan if temp == -1 else temp/1000)
+            is_KF_ran_atleast_once[0] = True
+            z_hat[0] = x_est[0][0]
+            v_hat[0] = x_est[1][0]
+            alpha_hat[0] = x_est[2][0]
+            beta_hat[0] = x_est[3][0]
+
+            KF_data[0] = z_hat[0]
+            KF_data[1] = v_hat[0]
+            KF_data[2] = alpha_hat[0]
+            KF_data[3] = beta_hat[0]
+
+            if use_altitude_hold[0]:
+                if not is_current_altitude_snap_shot_taken[0]:
+                    current_altitude_snap_shot_mts[0] = temp/1000
+                    var_e_RC_mid_percentage[0] = rc.trimmer_VRE_percentage()
+                    is_current_altitude_snap_shot_taken[0] = True
+            else:
+                is_current_altitude_snap_shot_taken[0] = False
+
+        if use_altitude_hold[0] and not is_drone_flying_close_to_ground[0]:
+            lqr.set_alpha_beta(alpha_hat[0], beta_hat[0])
+            lqr.set_gains(-gain_kp_from_rc[0], -gain_kd_from_rc[0])
+            throttle_ref_from_LQR[0] = lqr.control_action(np.array([[z_hat[0]], [v_hat[0]]]),
+                                                          reference_altitude_mts=altitude_ref_mts[0],
+                                                          recalculate_dynamics=True,
+                                                          pitch_rad=euler[1],
+                                                          roll_rad=euler[2])
+            throttle_ref_from_LQR[0] = max(
+                1000, min(throttle_ref_from_LQR[0]*900, 600) + 1000)
+            Tref_t[0] = (throttle_ref_from_LQR[0] - 1000)/900
+            print_debug(
+                f"alphan hat: {alpha_hat[0]} beta hat: {beta_hat[0]} k11: {-gain_kp_from_rc[0]} k12: {-gain_kd_from_rc[0]} Tref_LQR: {Tref_t[0]} alt_hat: {z_hat[0]} alt_ref: {altitude_ref_mts[0]}")
+
+    if enable_caching[0]:
+        time_before_thread_starts[0] = time_ns()
+
+    # initialisation checks: are the switches at the correct positions before flying?
```

Comment:
On the one hand, it looks like these initialisation checks work: the code gets stuck there until the switches are at the correct positions. However, it seems that we can un-kill and start to fly. I'm not sure why.